packages/image_picker/image_picker/example/ios/RunnerUITests/ImagePickerFromGalleryUITests.m

```diff
@@ -0,0 +1,170 @@
+  }
+
+  XCTAssertTrue(pickButton.exists);
+  [pickButton tap];
+
+  // There is a known bug where the permission popups interruption won't get fired until a tap
+  // happened in the app. We expect a permission popup so we do a tap here.
+  [self.app tap];
+
+  // Find and tap on the `Cancel` button.
+  NSPredicate* predicateToFindCancelButton =
+      [NSPredicate predicateWithFormat:@"label == %@", @"Cancel"];
+
+  XCUIElement* cancelButton =
+      [self.app.buttons elementMatchingPredicate:predicateToFindCancelButton];
+  if (![cancelButton waitForExistenceWithTimeout:30]) {
+    NSLog(@"%@", self.app.debugDescription);
+    XCTFail(@"Failed due to not able to find Cancel button with %@ seconds", @(30));
+  }
+
+  XCTAssertTrue(cancelButton.exists);
+  [cancelButton tap];
+
+  // Find the "not picked image text".
+  XCUIElement* imageNotPickedText = [self.app.otherElements
+      elementMatchingPredicate:[NSPredicate
+                                   predicateWithFormat:@"label == %@",
+                                                       @"You have not yet picked an image."]];
+  if (![imageNotPickedText waitForExistenceWithTimeout:30]) {
+    NSLog(@"%@", self.app.debugDescription);
+    XCTFail(@"Failed due to not able to find imageNotPickedText with %@ seconds", @(30));
+  }
+
+  XCTAssertTrue(imageNotPickedText.exists);
+}
+
+- (void)launchPickerAndPick {
+  // Find and tap on the pick from gallery button.
+  NSPredicate* predicateToFindImageFromGalleryButton =
+      [NSPredicate predicateWithFormat:@"label == %@", @"image_picker_example_from_gallery"];
+
+  XCUIElement* imageFromGalleryButton =
+      [self.app.otherElements elementMatchingPredicate:predicateToFindImageFromGalleryButton];
+  if (![imageFromGalleryButton waitForExistenceWithTimeout:30]) {
+    NSLog(@"%@", self.app.debugDescription);
+    XCTFail(@"Failed due to not able to find image from gallery button with %@ seconds", @(30));
+  }
+
+  XCTAssertTrue(imageFromGalleryButton.exists);
+  [imageFromGalleryButton tap];
+
+  // Find and tap on the `pick` button.
+  NSPredicate* predicateToFindPickButton =
+      [NSPredicate predicateWithFormat:@"label == %@", @"PICK"];
+
+  XCUIElement* pickButton = [self.app.buttons elementMatchingPredicate:predicateToFindPickButton];
+  if (![pickButton waitForExistenceWithTimeout:30]) {
+    NSLog(@"%@", self.app.debugDescription);
+    XCTFail(@"Failed due to not able to find pick button with %@ seconds", @(30));
+  }
+
+  XCTAssertTrue(pickButton.exists);
+  [pickButton tap];
+
+  // Find an image and tap on it. (IOS 14 UI, images are showing direclty)
+  XCUIElement* aImage = self.app.scrollViews.firstMatch.images.firstMatch;
+  XCUIElement* allPhotosCell = [self.app.cells
+      elementMatchingPredicate:[NSPredicate predicateWithFormat:@"label == %@", @"All Photos"]];
+  XCUIElement* firstExist = [RunnerUITestUtils waitForFirstExistence:@[ aImage, allPhotosCell ]
```

Comment:
This feels heavy (and the check should probably be scheduled on a runloop instead of the `while`).  Can you instead check `if (@available(iOS 14.0, *))` directly?

Reply:
Ha, ditto what @amirh said, didn't see his comment until I submitted.

Reply:
done